frontend/src/api/k8s/templates.ts

```diff
@@ -20,20 +21,23 @@ export const assembleTemplate = (body: string, namespace: string): TemplateKind
   const description = getDescriptionFromK8sResource(servingRuntime);
 
   if (!name) {
-    throw new Error('Template name is required');
+    throw new Error('Serving runtime name is required');
   }
 
   return {
     kind: 'Template',
     apiVersion: 'template.openshift.io/v1',
     metadata: {
-      name,
+      name, // TODO: generate a random template name?
+      // user could change the template with a new serving runtime
+      // and create an other template with the old serving runtime
+      // if we keep the name, there could be a conflict
       namespace,
       labels: {
         'opendatahub.io/dashboard': 'true',
       },
       annotations: {
-        description,
+        'openshift.io/description': description,
```

Comment:
@lucferbux I changed this to `openshift.io/description` to make it more aligned with the current design so we can use `getDescriptionFromK8sResource` method.